app/api/v1/admin/diagnoses.rb

```diff
@@ -0,0 +1,37 @@
+module V1
+  module Admin
+    class Diagnoses < API
+      desc 'create a new diagnosis'
+      # params do
+      #   requires :name, type: String
+      #   requires :symptom, type: String
+      #   requires :medication_list, type: String
+      # end
+
+      resources :diagnoses do
+        post do
+          Diagnosis.create!(params)
+        end
+
+        desc 'Update a specific diagnosis'
+        route_param :id do
+          put do
+            diagnosis = Diagnosis.find(params[:id])
+
+            diagnosis.update(params)
+            diagnosis
```

Comment:
```
Diagnosis.find(params[:id]).tap do |diagnosis|
  diagnosis.update!(params)
end
```

pls read about methods with a bang and what's the difference between `update` & `update!` and about `.tap`